realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy

```diff
@@ -93,33 +99,36 @@ class RealmTransformer extends Transform {
 
         // Find the model classes
         def realmObject = classPool.get('io.realm.RealmObject')
-        def modelClasses = classNames
+        def allModelClasses = allClassNames
                 .findAll { it.endsWith('RealmProxy') }
                 .collect { classPool.getCtClass(it).superclass }
                 .findAll { it.superclass?.equals(realmObject) }
-        logger.info "Model Classes: ${modelClasses*.name}"
+        logger.info "Model Classes: ${allModelClasses*.name}"
 
         // Populate a list of the fields that need to be managed with bytecode manipulation
-        def managedFields = []
-        modelClasses.each {
-            managedFields.addAll(it.declaredFields.findAll {
+        def allManagedFields = []
+        allModelClasses.each {
+            allManagedFields.addAll(it.declaredFields.findAll {
                 it.getAnnotations()
                 it.getAnnotation(Ignore.class) == null && !Modifier.isStatic(it.getModifiers())
             })
         }
-        logger.info "Managed Fields: ${managedFields*.name}"
+        logger.info "Managed Fields: ${allManagedFields*.name}"
 
         // Add accessors to the model classes
-        modelClasses.each {
+        allModelClasses.each {
```

Comment:
I don't think we can split the `ClassPool` since it is not just a `Set` of classes. `ClassPool` must contain all class paths to load classes in application project.